### PR TITLE
GA release 4.0.0

### PR DESCRIPTION
**Pull request by rodireich**
Release source-mssql 4.0.0 as a breaking change letting user opt-in

### CODE REVIEW

**Review by evantahler — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-mssql/metadata.yaml [hunk 1]
```diff
@@ -32,6 +28,9 @@ data:
     - language:python
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-mssql/metadata.yaml:28]: It's not python!

File: airbyte-integrations/connectors/source-mssql/metadata.yaml [hunk 2]
```diff
@@ -32,6 +28,9 @@ data:
...
   releases:
     breakingChanges:
+      4.0.0:
+        message: "Beta release of certified source mssql connector."
+        upgradeDeadline: "2024-04-07"
```
[the remark this answers is not in the document]
Comment (rodireich) [reply airbyte-integrations/connectors/source-mssql/metadata.yaml:33]: Right, it's GA really.

File: docs/integrations/sources/mssql.md [hunk 3]
```diff
@@ -342,8 +327,9 @@ WHERE actor_definition_id ='b5ea17b1-f170-46dc-bc31-cc744ca984c1' AND (configura
 
 | Version | Date       | Pull Request                                                                                                      | Subject                                                                                                                                         |
 |:--------|:-----------|:------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------|
+| 4.0.0   | 2024-03-06 | [35873](https://github.com/airbytehq/airbyte/pull/35873)                                                          | Beta release.                                                                                                                                   |
```
[the remark this answers is not in the document]
Comment (rodireich) [reply docs/integrations/sources/mssql.md]: I removed mentions of Beta and added details.

File: airbyte-integrations/connectors/source-mssql/metadata.yaml [hunk 4]
```diff
@@ -18,20 +18,18 @@ data:
...
-  releaseStage: alpha
+  releaseStage: generally_available
   supportLevel: community
   tags:
     - language:java
-    - language:python
   releases:
     breakingChanges:
+      4.0.0:
+        message: "Certified source mssql connector."
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-mssql/metadata.yaml:31]: If you say certified here, then you need to change line 25 too

File: airbyte-integrations/connectors/source-mssql/metadata.yaml [hunk 5]
```diff
@@ -18,20 +18,18 @@ data:
...
-      # CI pipeline is broken for mssql
-      dockerImageTag: 3.6.1
       enabled: true
     oss:
-      # CI pipeline is broken for mssql
-      dockerImageTag: 3.6.1
       enabled: true
-  releaseStage: alpha
+  releaseStage: generally_available
   supportLevel: community
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-mssql/metadata.yaml:25]: this is very confusing that we are highlighting it as "certified' but keeping it community

File: airbyte-integrations/connectors/source-mssql/metadata.yaml [hunk 6]
```diff
@@ -18,20 +18,18 @@ data:
...
   tags:
     - language:java
-    - language:python
   releases:
     breakingChanges:
+      4.0.0:
+        message: "Certified source mssql connector."
+        upgradeDeadline: "2024-04-07"
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-mssql/metadata.yaml:32]: should we be using scopedImpact? https://docs.airbyte.com/connector-development/connector-metadata-file/#scopedimpact

File: docs/integrations/sources/mssql.md [hunk 7]
```diff
@@ -1,4 +1,9 @@
 # Microsoft SQL Server (MSSQL)
+Airbyte's certified MySQL connector offers the following features:
+* Multiple methods of keeping your data fresh, including [Change Data Capture (CDC)](https://docs.airbyte.com/understanding-airbyte/cdc) using the [binlog](https://dev.mysql.com/doc/refman/8.0/en/binary-log.html).
+* All available [sync modes](https://docs.airbyte.com/cloud/core-concepts#connection-sync-modes), providing flexibility in how data is delivered to your destination.
```
Comment (katmarkham) [inline docs/integrations/sources/mssql.md:4]: let's be specific here about sync mode types

File: docs/integrations/sources/mssql.md [hunk 8]
```diff
@@ -41,13 +35,14 @@ This is dependent on your networking setup. The easiest way to verify if Airbyte
 
 This step is optional but highly recommended to allow for better permission control and auditing. Alternatively, you can use Airbyte with an existing user in your database.
 
-_Coming soon: suggestions on how to create this user._
-
 #### 3. Your database user should now be ready for use with Airbyte!
 
+#### Airbyte Cloud
+On Airbyte Cloud, only secured connections to your MSSQL instance are supported in source configuration, Either by configuring you connection usign one of the supported SSL Methods or by using an SSH Tunnel.
```
Comment (katmarkham) [inline docs/integrations/sources/mssql.md:41]: Spelling errors
On Airbyte Cloud, only secured connections to your MSSQL instance are supported in source configuration. You may either configure your connection using one of the supported SSL Methods or by using an SSH Tunnel.

File: docs/integrations/sources/mssql.md [hunk 9]
```diff
@@ -41,13 +35,14 @@ This is dependent on your networking setup. The easiest way to verify if Airbyte
...
+
 ## Change Data Capture \(CDC\)
 
-We use [SQL Server's change data capture feature](https://docs.microsoft.com/en-us/sql/relational-databases/track-changes/about-change-data-capture-sql-server?view=sql-server-2017) to capture row-level `INSERT`, `UPDATE` and `DELETE` operations that occur on cdc-enabled tables.
+We use [SQL Server's change data capture feature](https://docs.microsoft.com/en-us/sql/relational-databases/track-changes/about-change-data-capture-sql-server?view=sql-server-2017) with transaction logs to capture row-level `INSERT`, `UPDATE` and `DELETE` operations that occur on cdc-enabled tables.
```
Comment (katmarkham) [inline docs/integrations/sources/mssql.md:45]: capitalize CDC

File: docs/integrations/sources/mssql.md [hunk 10]
```diff
@@ -60,27 +55,17 @@ Please read the [CDC docs](../../understanding-airbyte/cdc.md) for an overview o
...
-| Snapshot Isolation Level |      Enum: `Snapshot`, `Read Committed`      |     `Snapshot`     | Mode to control which transaction isolation level is used and how long the connector locks tables that are designated for capture. If you don't know which one to choose, just use the default one. See documentation [here](https://debezium.io/documentation/reference/stable/connectors/sqlserver.html#sqlserver-property-snapshot-isolation-mode) for details. Under the hood, this parameter sets the `snapshot.isolation.mode` in Debezium.         |
-
 #### CDC Limitations
 
 - Make sure to read our [CDC docs](../../understanding-airbyte/cdc.md) to see limitations that impact all databases using CDC replication.
-- There are some critical issues regarding certain datatypes. Please find detailed info in [this Github issue](https://github.com/airbytehq/airbyte/issues/4542).
+- `hierarchyid` and `sql_variant` types are not processed in CDC migration type (not supported by debezium). For more details please check
```
Comment (katmarkham) [inline docs/integrations/sources/mssql.md:61]: Debezium

File: docs/integrations/sources/mssql.md [hunk 11]
```diff
@@ -342,8 +327,9 @@ WHERE actor_definition_id ='b5ea17b1-f170-46dc-bc31-cc744ca984c1' AND (configura
 
 | Version | Date       | Pull Request                                                                                                      | Subject                                                                                                                                         |
 |:--------|:-----------|:------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------|
+| 4.0.0   | 2024-03-06 | [35873](https://github.com/airbytehq/airbyte/pull/35873)                                                          | Terabyte sized tables support, reliability improvements, bug fixes.                                                                             |
```
Comment (katmarkham) [inline docs/integrations/sources/mssql.md:330]: Terabyte-sized tables support, reliability improvements, bug fixes.

File: airbyte-integrations/connectors/source-mssql/metadata.yaml (hunk 4 above)
Comment (katmarkham) [reply airbyte-integrations/connectors/source-mssql/metadata.yaml:31]: might want to write something along the lines of
"We have overhauled our MSSQL source connector and it is now supported by the Airbyte team!  To benefit from  new features, including terabyte-sized table support, reliability improvements, expanded datetime data types, and various bug fixes, please opt in to the 4.0.0 version."

File: docs/integrations/sources/mssql-migrations.md [hunk 12]
```diff
@@ -1,5 +1,9 @@
 # Microsoft SQL Server (MSSQL) Migration Guide
 
+## Upgrading to 4.0.0
+Source MSSQL provides Incremental sync that can read unlimited sized table and can resume if initial read has failed.
```
Comment (katmarkham) [inline docs/integrations/sources/mssql-migrations.md:4]: Source MSSQL provides incremental sync that can read unlimited sized tables and can resume if the initial read has failed.
Upgrading from previous versions will be seamless and does not require any intervention.

File: docs/integrations/sources/mssql.md [hunk 13]
```diff
@@ -1,4 +1,9 @@
 # Microsoft SQL Server (MSSQL)
+Airbyte's certified MySQL connector offers the following features:
```
Comment (katmarkham) [inline docs/integrations/sources/mssql.md:2]: Airbyte's certified MSSQL connector offers the following features:

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-mssql/metadata.yaml (hunk 5 above)
Comment (rodireich) [reply airbyte-integrations/connectors/source-mssql/metadata.yaml:25]: thanks @katmarkham . I understand that
But we're on our way to certified.
If GA works well and no major issues found we will officially certify it.

File: airbyte-integrations/connectors/source-mssql/metadata.yaml (hunk 6 above)
Comment (rodireich) [reply airbyte-integrations/connectors/source-mssql/metadata.yaml:32]: I don't think `scopedImpact` is applicable here. 
There's no stream name that is breaking

File: docs/integrations/sources/mssql.md (hunk 13 above)
Comment (rodireich) [reply docs/integrations/sources/mssql.md:2]: 😳

File: docs/integrations/sources/mssql.md (hunk 8 above)
Comment (rodireich) [reply docs/integrations/sources/mssql.md:41]: Thank you!

**Review by evantahler — APPROVED** [review]

**Review by katmarkham — APPROVED** [review]

**Pull request merged**